offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
+          [ HH.div [ mkClass "column is-narrow" ]
+              [ HH.button [ mkClass $ "button is-rounded is-medium is-responsive danaswap-btn-has-border" ] [ HH.text "SUBMIT" ]
+              ]
+          ]
+      ]
+
+    renderAddLiquidityForm (Pool ({ assetA: (Asset a), assetB: (Asset b) })) = HH.div [ mkClass "mt-6" ]
+      [ HH.div [ mkClass "columns is-mobile is-vcentered" ]
+          [ HH.div [ mkClass "column is-3" ]
+              [ HH.div [ mkClass "field" ]
+                  [ HH.div [ mkClass "control" ] [ HH.input [ mkClass "input is-rounded" ] ]
+                  ]
+              ]
+          , HH.div [ mkClass "column is-1" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch" ] [ HH.text a.name ]
+              ]
+          ]
+      , HH.div [ mkClass "columns is-mobile is-vcentered" ]
+          [ HH.div [ mkClass "column is-3" ]
+              [ HH.div [ mkClass "field" ]
+                  [ HH.div [ mkClass "control" ] [ HH.input [ mkClass "input is-rounded" ] ]
+                  ]
+              ]
+          , HH.div [ mkClass "column is-1" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch" ] [ HH.text b.name ]
+              ]
+          ]
+      , HH.p [ mkClass "subtitle is-6 my-0" ]
+          [ HH.text "Liquidity Tokens: 0"
+          ]
+      , HH.p [ mkClass "subtitle is-6" ]
+          [ HH.text "Fee: 2.3 ADA"
+          ]
+      , HH.div [ mkClass "columns" ]
+          [ HH.div [ mkClass "column is-narrow" ]
+              [ HH.button [ mkClass $ "button is-rounded is-medium is-responsive danaswap-btn-has-border" ] [ HH.text "SUBMIT" ]
+              ]
+          ]
       ]
+
+    renderWithdrawLiquidityForm = HH.div [ mkClass "mt-6" ]
+      [ HH.div [ mkClass "columns is-mobile is-vcentered" ]
+          [ HH.div [ mkClass "column is-3" ]
+              [ HH.div [ mkClass "field" ]
+                  [ HH.div [ mkClass "control" ] [ HH.input [ mkClass "input is-rounded" ] ]
+                  ]
+              ]
+          , HH.div [ mkClass "column is-2" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch" ] [ HH.text "Liquidity Tokens" ]
+              ]
+          ]
+      , HH.p [ mkClass "subtitle is-6 my-0" ]
+          [ HH.text "ADA Payout: 0"
+          ]
+      , HH.p [ mkClass "subtitle is-6 my-0" ]
+          [ HH.text "DANA Payout: 0"
+          ]
+      , HH.p [ mkClass "subtitle is-6 my-0" ]
+          [ HH.text "Slippage: 4%"
+          ]
+      , HH.p [ mkClass "subtitle is-6" ]
+          [ HH.text "Fee: 2.3 ADA"
+          ]
+      , HH.div [ mkClass "columns" ]
+          [ HH.div [ mkClass "column is-narrow" ]
+              [ HH.button [ mkClass $ "button is-rounded is-medium is-responsive danaswap-btn-has-border" ] [ HH.text "SUBMIT" ]
+              ]
+          ]
+      ]
+
+mkClass :: forall (r :: Row Type) (i :: Type). String -> HP.IProp (class :: String | r) i
+mkClass = HP.class_ <<< HH.ClassName
```

Comment:
We had a rule at my old gig that it was always better to use `HP.classes [ HH.ClassName "..", HH.ClassName ".." ]` for long-term maintenance. I wish I remember the argument better, but they had me convinced be the end. This system eliminates safety too as it *was* `newtype`'d but is now stringly-typed.

Reply:
indeed, but I reckon it might be overkill here